rust/headless-client/src/dns_control/windows.rs

```diff
@@ -46,20 +43,19 @@ impl DnsController {
     ///
     /// Must be async to match the Linux signature
     #[allow(clippy::unused_async)]
-    pub(crate) async fn set_dns(&mut self, dns_config: &[IpAddr]) -> Result<()> {
-        deactivate().context("Failed to deactivate DNS control")?;
```

Comment:
Since we're setting the NRPT rule in the registry directly, we can also modify it in place. With Powershell we had been deleting and re-adding the rule, since modifying it required a third command that was difficult to set up.

So since we can modify it in-place, this seals #5879